File: SpringBootConfigurable/src/test/java/com/example/java/AccountTest.java
package com.example.java;

import com.example.java.account.Account;
import com.example.java.config.ConfigurableConfig;
import com.example.java.config.SecurityConfig;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
@Transactional
@Import({ConfigurableConfig.class, SecurityConfig.class})
public class AccountTest {

    @PersistenceContext
    private EntityManager entityManager;

    @DisplayName("Account 비밀번호 리셋 테스트")
    @Test
    public void test() {
        // given
        Account account = new Account("heowc", "1234");
        entityManager.persist(account);
        entityManager.flush();
        entityManager.clear();

        // when
        Account heowc = entityManager.find(Account.class, account.getId());
        heowc.resetPassword();
        entityManager.flush();
        entityManager.clear();

        // then
        Account result = entityManager.find(Account.class, account.getId());
        assertThat(result.getId()).isEqualTo(account.getId());
        assertThat(result.getPassword()).isNotEqualTo(account.getPassword());
    }
}

File: SpringBootConfigurable/src/test/kotlin/com/example/kotlin/AccountTest.kt
package com.example.kotlin

import com.example.kotlin.account.Account
import com.example.kotlin.config.ConfigurableConfig
import com.example.kotlin.config.SecurityConfig
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.transaction.annotation.Transactional
import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext

@SpringBootTest
@Transactional
@Import(ConfigurableConfig::class, SecurityConfig::class)
class AccountTest {

    @PersistenceContext
    private lateinit var entityManager: EntityManager

    @Test
    fun `Account 비밀번호 리셋 테스트`() {
        // given
        val account = Account("heowc", "1234")
        entityManager.persist(account)
        entityManager.flush()
        entityManager.clear()

        // when
        val heowc = entityManager.find(Account::class.java, account.id)
        heowc.resetPassword()
        entityManager.flush()
        entityManager.clear()

        // then
        val result = entityManager.find(Account::class.java, account.id)
        assertThat(result.id).isEqualTo(account.id)
        assertThat(result.password).isNotEqualTo(account.password)
    }
}
